lib/deployer.rb

```diff
@@ -24,53 +24,53 @@ def initialize(environment:, repos:, tag: nil, before_command: nil)
     )
     @tag = tag
     ensure_tag_present_in_all_repos! if tag
-    prompt_user_for_branch_confirmation!
-    prompt_user_for_approval_confirmation!
   end
 
   # rubocop:disable Metrics/CyclomaticComplexity
   def deploy_all
-    puts "Repositories: #{repos.map(&:name).join(', ')}"
+    render_markdown('***')
+    render_markdown("# Deploying the following repositories (#{tag || 'default branch'})")
+    render_markdown(repos.map { |repo| "* #{repo.name}" }.join("\n"))
+
+    prompt_user_for_branch_confirmation!
+    prompt_user_for_approval_confirmation!
+
     progress_bar.start
 
     results = Parallel.map(
       repos,
       in_processes: Settings.num_parallel_processes,
-      finish: lambda do |repo, _i, _result|
-        if Settings.progress_file.enabled
-          filename = File.join(Settings.progress_file.location, "#{File.basename(repo.name)}-deploy.log")
-          File.write(filename, "#{Time.now} : #{repo.name} deploy complete")
+      finish: lambda do |repo, _i, result_array|
+        result_array.each do |result|
+          progress_bar.log(log_result(result:))
         end
         progress_bar.advance(repo: repo.name)
+        next unless Settings.progress_file.enabled
+
+        filename = File.join(Settings.progress_file.location, "#{File.basename(repo.name)}-deploy.log")
+        File.write(filename, "#{Time.now} : #{repo.name} deploy complete")
       end
     ) do |repo|
       within_project_dir(repo:, environment:) do |env|
         auditor.audit(repo: repo.name)
         run_before_command!(env)
         set_deploy_target!
         status, output = deploy(env)
-        Result.new(
-          env == environment ? repo.name : "#{repo.name} (#{env})",
-          env,
-          status ? colorize_success('success') : colorize_failure('FAILED'),
-          output
-        )
+        Result.new(repo.name, env, status, output)
       end
     end.flatten
 
     auditor.report
 
-    build_report_table!(results)
-
-    puts report_table.render(:unicode)
```

Comment:
Replace the table-based output for deploys with log-based output (see screenshots in PR description)